File: backend/app/schemas/diadoc.py
from pydantic import BaseModel
from typing import Optional
import uuid
from datetime import datetime


class DiadocCheckRequest(BaseModel):
    document_id: uuid.UUID


class DiadocCheckResponse(BaseModel):
    ok: bool
    errors: list[str] = []
    warnings: list[str] = []


class DiadocSendRequest(BaseModel):
    document_id: uuid.UUID
    counterparty_inn: str


class DiadocSendResponse(BaseModel):
    message_id: str
    entity_id: str
    status: str


class DiadocStatusResponse(BaseModel):
    id: uuid.UUID
    document_id: uuid.UUID
    message_id: str
    entity_id: str
    status: str
    ai_check_result: dict
    counterparty_inn: str
    error_message: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DiadocQueueItem(BaseModel):
    id: uuid.UUID
    document_id: uuid.UUID
    status: str
    counterparty_inn: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/diadoc.ts
export interface DiadocCheckResult {
  ok: boolean;
  errors: string[];
  warnings: string[];
}

export interface DiadocQueueItem {
  id: string;
  document_id: string;
  status: string;
  counterparty_inn: string;
  created_at: string;
}

export interface DiadocStatusRecord {
  id: string;
  document_id: string;
  message_id: string;
  entity_id: string;
  status: string;
  ai_check_result: Record<string, unknown>;
  counterparty_inn: string;
  error_message: string;
  created_at: string;
  updated_at: string;
}
